source/2_platform/rex_windows/src/gui_application.cpp

```diff
@@ -15,12 +15,46 @@
 
 #include <Windows.h>
 
-// NOLINTBEGIN(modernize-use-nullptr,-warnings-as-errors)
+// NOLINTBEGIN(cppcoreguidelines-pro-type-union-access)
+// NOLINTBEGIN(modernize-use-nullptr)
 
 namespace rex
 {
   namespace win32
   {
+    class GuiApplicationState
+    {
+    public:
+      enum class Type
+      {
+        Resizing  = BIT(0),
+        Minimized = BIT(1),
+        Maximized = BIT(2)
+      };
+
+      GuiApplicationState()
+          : m_state(0u)
+      {
+      }
+
+      void add_state(GuiApplicationState::Type state)
+      {
+        m_state = m_state & static_cast<u32>(state);
+      }
+      void remove_state(GuiApplicationState::Type state)
+      {
+        m_state = m_state & ~static_cast<u32>(state);
+      }
+
+      bool has_state(GuiApplicationState::Type state) const
```

Comment:
this already exists. rsl::has_flag

Reply:
This is not present in rsl I checked.
RSL expects the first argument to be an integer and the second to be an enum which is the other way around here.